qiskit/quantum_info/synthesis/one_qubit_decompose.py

```diff
@@ -464,50 +517,90 @@ def _circuit_u(theta, phi, lam, phase, simplify=True, atol=DEFAULT_ATOL):
         return circuit
 
     @staticmethod
-    def _circuit_psx_gen(theta, phi, lam, phase, atol, pfun, xfun, xpifun=None):
+    def _circuit_psx_gen(
+        theta,
+        phi,
+        lam,
+        phase,
+        atol,
+        pfun,
+        xfun,
+        xpifun=None,
+        xfun_fid=1.0,
+        xpifun_fid=1.0,
+        simplify=True,
+    ):
         """
         Generic X90, phase decomposition
 
         NOTE: `pfun` is responsible for eliding gates where appropriate (e.g., at angle value 0).
         """
         qr = QuantumRegister(1, "qr")
         circuit = QuantumCircuit(qr, global_phase=phase)
-        # Early return for zero SX decomposition
-        if np.abs(theta) < atol:
+
+        if not simplify:
+            atol = -1.0
+
+        # infidelity estimates for each of the registered circuit types
+        # NOTE: When the infidelity estimates overlap, we're relying on alphabetical sorting.
```

Comment:
Alphabetical sorting feels janky but I guess it gives the correct order: empty, one sx, one x, two sx and is self-contained and well-commented so I allow it.

Reply:
It's goofy for sure, and it is courting trouble. I'd be more fussed if it were doing something more important than capturing our personal preference among circuit types when the mathematics itself expresses no preference. In that light, it feels like any bugs it might introduce are pretty benign.